runtime/sema/check_remove_statement.go

```diff
@@ -31,6 +31,8 @@ func (checker *Checker) VisitRemoveStatement(statement *ast.RemoveStatement) (_
 		})
 	}
 
+	checker.ObserveImpureOperation(statement)
```

Comment:
Maybe it's worth verifying checking of all statements and expressions to see if there are similar cases in the code base.